src/electionguard/elgamal.cpp

```diff
@@ -267,21 +276,25 @@ namespace electionguard
         return make_unique<ElGamalCiphertext>(move(pad), move(data));
     }
 
-    unique_ptr<ElGamalCiphertext> elgamalEncrypt(uint64_t m,
-                                                 const TwoTriplesAndAQuadruple &precomputedValues)
+    unique_ptr<ElGamalCiphertext> elgamalEncrypt(uint64_t m, const ElementModP &publicKey,
+                                                 const PrecomputedEncryption &precomputedValues)
     {
-        auto triple1 = precomputedValues.get_triple1();
+        // (g^R mod p, K^V ·K^R mod p) = (g^R mod p, K^(V+R) mod p)
 
-        auto pad = triple1->clone_g_to_exp();             // g^r
-        auto pubkey_pow_r = triple1->get_pubkey_to_exp(); // K^r
+        auto pad = precomputedValues.getPad()->clone();            // g^r mod p
+        auto pubkey_pow_r = precomputedValues.getBlindingFactor(); // K^r mod p
         unique_ptr<ElementModP> data = nullptr;
         if (m == 1) {
-            data = mul_mod_p(G(), *pubkey_pow_r);
+            data = mul_mod_p(publicKey, *pubkey_pow_r); // K^1 * K^r mod p
+        } else if (m == 0) {
+            data = pubkey_pow_r->clone(); // K^0 * K^r mod p
         } else {
-            data = pubkey_pow_r->clone();
+            auto pubkey_pow_m = pow_mod_p(publicKey, m);
+            data = mul_mod_p(publicKey, *pubkey_pow_r); // K^V * K^r mod p
```

Comment:
i considered throwing here but instead allow it to pass since cumulative voting is a thing and its called out in the spec. we still wont support it however since the disjunctive proof requires a proof of zero or 1, but the encryption will work.